File: exercicios/R/2003/2003.r
input = file('stdin','r')

while(TRUE){ 
    line = strsplit(readLines(input,n=1), ":")
    hmax = as.integer(line[[1]][1])
    mmax = as.integer(line[[1]][2])

    hh = as.integer((hmax+1) - 8)

    if(hh < 0){
        write("Atraso maximo: 0",'')
    }else{
        mmax = mmax + 60 * hh
        write(sprintf("Atraso maximo: %d",mmax),'')
    }

    if(length(line) == 0){
        break;}
}

File: exercicios/JavaScript/2003/2003.js
var input = require("fs").readFileSync("stdin",'utf8');
var lines = input.split('\n')

console.log(lines.length)
while(lines.length > 0){
    var v = lines.shift().split(':')

    var hmax = parseInt(v[0]) + 1, mmax = parseInt(v[1])

    var hh = hmax - 8
 
    if(hh < 0){
        console.log('Atraso maximo: 0')
    }else{
        mmax += 60 * hh
        console.log('Atraso maximo: ' + mmax)
    }
}
